recurring_payments_stripe/models/account_move.py

```diff
@@ -0,0 +1,94 @@
+import logging
+
+import stripe
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_register_payment(self):
+        """
+        Override `action_register_payment` to automatically process Stripe
+        payment on subscriptions.
+        """
+        for invoice in self:
+            # Find the subscription associated with the invoice, if it exists
+            subscription = invoice.subscription_id
+
+            # Check if the subscription is recurring and has a payment method
+            if subscription and subscription.charge_automatically:
+                provider = subscription.provider_id
+                stripe.api_key = provider.stripe_secret_key
+                token = self.env["payment.token"].search(
+                    [("provider_id", "=", provider.id)]
+                )
+                try:
+                    # Create the PaymentIntent and confirm it immediately
+                    payment_intent = stripe.PaymentIntent.create(
+                        # Stripe usa centavos
+                        amount=int(invoice.amount_total * 100),
+                        currency=invoice.currency_id.name.lower(),
+                        customer=token.provider_ref,
+                        payment_method=token.stripe_payment_method,
+                        # Para pagos automáticos sin intervención del usuario
+                        off_session=True,
+                        # Confirmar el PaymentIntent inmediatamente
+                        confirm=True,
+                        metadata={"odoo_invoice_id": str(invoice.id)},
+                    )
+
+                    # Manejar el resultado del PaymentIntent
```

Comment:
Comments should be in English.

Reply:
Fixed translation problem in the comment